.github/workflows/goBuild.yml

```diff
@@ -4,20 +4,43 @@ on:
       os-dependencies:
         required: false
         type: string
+      only-latest-golang:
+        required: false
+        type: boolean
+        default: false
     secrets:
       SSH_PRIVATE_KEY:
         required: false
       PAT:
         required: false
 
 jobs:
+  set-go-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-output.outputs.matrix }}
+    steps:
+      - name: Set go-matrix latest version
+        id: set-matrix-latest
+        if: inputs.only-latest-golang
+        run: |
+          echo "matrix={\"include\":[{\"go\":\"1.19\"}]}" >> ${GITHUB_ENV}
+      - name: Set default go-matrix
+        id: set-matrix-default
+        if: inputs.only-latest-golang == false
+        run: |
+          echo "matrix={\"include\":[{\"go\":\"1.18\"},{\"go\":\"1.19\"}]}" >> ${GITHUB_ENV}
+      - id: set-output
+        run: |
+          echo "matrix=${{ env.matrix }}" >> ${GITHUB_OUTPUT}
+
   build:
+    needs: set-go-matrix
     runs-on: ubuntu-latest
     env:
       GOPRIVATE: go.step.sm
     strategy:
-      matrix:
-        go: ['1.18', '1.19']
+      matrix: ${{ fromJson(needs.set-go-matrix.outputs.matrix) }}
```

Comment:
The value of matrix seems to be {\"include\":[{\"go\":\"1.19\"}]}, is that include ok?

Reply:
See above answer.